site/content/docs/01-component-format.md

```diff
@@ -0,0 +1,188 @@
+---
+title: Component format
+---
+
+---
+
+Components are the building blocks of Svelte applications. They are written into `.svelte` files, using a superset of HTML.
+
+All three sections — script, styles and markup — are optional.
+
+```html
+<script>
+	// logic goes here
+</script>
+
+<style>
+	/* styles go here */
+</style>
+
+<!-- markup (zero or more items) goes here -->
+```
+
+### &lt;script&gt;
+
+A `<script>` block contains JavaScript that runs when a component instance is created. Variables declared (or imported) at the top level are 'visible' from the component's markup. There are four additional rules:
+
+#### 1. `export` creates a component prop
+
+---
+
+Svelte uses the `export` keyword to mark a variable declaration as a *property* or *prop*, which means it becomes accessible to consumers of the component:
+
+```html
+<script>
+	// these properties can be set externally
+	export let foo;
+	export let bar = 'optional default value';
+
+	// Values that are passed in as props
+	// are immediately available
+	console.log(foo, bar);
+
+	// function declarations cannot be set externally,
+	// but can be accessed from outside
+	export function instanceMethod() {
+		alert(foo);
+	}
+</script>
+```
+
+#### 2. Assignments are 'reactive'
+
+---
+
+To change component state and trigger a re-render, just assign to a locally declared variable.
+
+Update expressions (`count += 1`) and property assignments (`obj.x = y`) have the same effect.
+
+```html
+<script>
+	let count = 0;
+
+	function handleClick () {
+		// calling this function will trigger a re-render
+		// if the markup references `count`
+		count = count + 1;
+	}
+</script>
+```
+
+#### 3. `$:` marks a statement as reactive
+
+---
+
+Any top-level statement (i.e. not inside a block or a function) can be made reactive by prefixing it with the `$:` label. Reactive statements run immediately before the component updates, whenever the values that they depend on have changed.
+
+```html
+<script>
+	export let title;
+
+	// this will update `document.title` whenever
+	// the `title` prop changes
+	$: document.title = title;
+
+	$: {
+		console.log(`multiple statements can be combined`);
+		console.log(`the current title is ${title}`);
+	}
+</script>
+```
+
+---
+
+If a statement consists entirely of an assignment to an undeclared variable, Svelte will inject a `let` declaration on your behalf.
+
+```html
+<script>
+	export let num;
+	$: squared = num * num;
+	$: cubed = squared * num;
+</script>
+```
+
+#### 4. Prefix stores with `$` to access their values
+
+---
+
+Any time you have a reference to a store, you can access its value inside a component by prefixing it with the `$` character. This causes Svelte to declare the prefixed variable, and set up a store subscription that will be unsubscribed when appropriate.
```

Comment:
Do we want to mention here (and if not here, where?) that this is only for stores that are defined at the top level?

Reply:
yes. done